ui/src/app/machines/views/MachineDetails/MachineNetwork/NetworkFields/NetworkFields.test.tsx

```diff
@@ -307,77 +414,158 @@ describe("NetworkFields", () => {
         </MemoryRouter>
       </Provider>
     );
-    await changeField(wrapper, "SubnetSelect select", "subnet", 2);
-    wrapper.update();
-    expect(wrapper.find("LinkModeSelect").exists()).toBe(true);
-    expect(wrapper.find("LinkModeSelect select").prop("value")).toBe(
-      NetworkLinkMode.AUTO
+    fireEvent.change(
+      screen.getByRole("combobox", { name: SubnetSelectLabel.Select }),
+      {
+        target: { value: state.subnet.items[1].id.toString() },
+      }
+    );
+    expect(
+      screen.getByRole("combobox", { name: LinkModeSelectLabel.Select })
+    ).toBeInTheDocument();
+    const linkModeSelect = screen.getByRole("combobox", {
+      name: LinkModeSelectLabel.Select,
+    });
+    await waitFor(() =>
+      expect(
+        within(linkModeSelect).getByRole("option", { selected: true })
+      ).toHaveAttribute("value", NetworkLinkMode.AUTO)
+    );
+  });
+
+  it("reset the mode field to 'unconfigured' when editing and removed subnet", async () => {
+    const store = mockStore(state);
+    const onSubmit = jest.fn();
+    render(
+      <Provider store={store}>
+        <MemoryRouter
+          initialEntries={[{ pathname: "/machine/abc123", key: "testKey" }]}
+        >
+          <Formik
+            initialValues={{
+              ...networkFieldsInitialValues,
+              mode: NetworkLinkMode.AUTO,
+            }}
+            onSubmit={onSubmit}
+          >
+            {({ handleSubmit }) => (
+              <form onSubmit={handleSubmit} aria-label="test form">
+                <NetworkFields
+                  interfaceType={NetworkInterfaceTypes.PHYSICAL}
+                  editing
+                />
+              </form>
+            )}
+          </Formik>
+        </MemoryRouter>
+      </Provider>
+    );
+    // Remove the subnet.
+    fireEvent.change(
+      screen.getByRole("combobox", { name: SubnetSelectLabel.Select }),
+      {
+        target: { value: "" },
+      }
+    );
+    expect(
+      screen.queryByRole("combobox", { name: LinkModeSelectLabel.Select })
+    ).not.toBeInTheDocument();
+    fireEvent.submit(screen.getByRole("form", { name: "test form" }));
+    await waitFor(() =>
+      expect(onSubmit.mock.calls[0][0].mode).toBe(NetworkLinkMode.LINK_UP)
```

Comment:
Apologies for the noise (I regret doing the RTL migration in this PR). This test is the only change in this file that is not a migration.